rust_src/src/fns.rs

```diff
@@ -306,4 +309,43 @@ pub extern "C" fn internal_equal_misc(
     }
 }
 
+/// Return list of 1 minute, 5 minute and 15 minute load averages.
+///
+/// Each of the three load averages is multiplied by 100, then converted
+/// to integer.
+///
+/// When USE-FLOATS is non-nil, floats will be used instead of integers.
+/// These floats are not multiplied by 100.
+///
+/// If the 5-minute or 15-minute load averages are not available, return a
+/// shortened list, containing only those averages which are available.
+///
+/// An error is thrown if the load average can't be obtained.  In some
+/// cases making it work would require Emacs being installed setuid or
+/// setgid so that it can read kernel information, and that usually isn't
+/// advisable.
+#[lisp_fn(min = "0")]
+pub fn load_average(use_floats: bool) -> Vec<LispObject> {
+    let mut load_avg: [libc::c_double; 3] = [0.0, 0.0, 0.0];
+    let loads = if cfg!(windows) {
```

Comment:
This style does not pass clippy.

What you need to do is make a `getloadavg` function which compiles to either the remacs_sys version or the libc version and call that function here.

Reply:
With all the same rustup, cargo and rustc versions I see at https://travis-ci.org/Wilfred/remacs/jobs/469420967 I `rustup component add clippy-preview` but running `make clippy` does not show me the same error. I've pushed up something that should be OK and I'll see if the Travis output agrees.

Reply:
Success!